packages/protoc-gen-ng/src/output/types/fields/map-message-field.ts

```diff
@@ -30,13 +30,15 @@ export class MapMessageField implements MessageField {
 
   printDeserializeBinaryFromReader(printer: Printer) {
     const msgVarName = `msg_${this.messageField.number}`;
+    const isStringKey = this.keyField.type === ProtoMessageFieldType.string || isNumberString(this.keyField);
```

Comment:
Not sure if this was a bug or intentional, but when deserializing a boolKeyMap, you get `{ true: 'true' }` without this instead of input `{ 1: 'true' }`

Reply:
This is definitely a bug. I guess this was casted to string at some moment before, and then was turned off resulting in the behavior you see